src/test/java/com/google/cloud/tools/maven/AbstractSingleYamlDeployMojoTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.maven;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.tools.appengine.api.deploy.AppEngineDeployment;
+import com.google.cloud.tools.appengine.api.deploy.AppEngineFlexibleStaging;
+import com.google.cloud.tools.appengine.api.deploy.AppEngineStandardStaging;
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+
+@RunWith(MockitoJUnitRunner.class)
+public abstract class AbstractSingleYamlDeployMojoTest<M extends AbstractSingleYamlDeployMojo> {
```

Comment:
Obviously this is personal preference, but can we write this as a test fixture instead of a superclass? I find it easier to trace through and debug tests that way, and I'm not sure how I feel about `@Test` inheritance from a readability standpoint.

I'm trying to figure out how to exactly do this, but perhaps DeployCronMojo would get the common code by instantiating SingleYamlDeployMojoTestFixture<M>(tmpDir) which would mock all required things (and probably have to call initMocks on itself). I honestly have no idea. Do you think this makes sense though?